src/command/lsp/mod.rs

```diff
@@ -0,0 +1,230 @@
+        Ok(RoverOutput::EmptySuccess)
+    }
+}
+
+async fn run_lsp(client_config: StudioClientConfig, lsp_opts: LspOpts) -> RoverResult<()> {
+    //TODO: Check this error handling is right i.e. if we don't get a supergraph.yaml passed
+    // this should fail rather than doing something else.
+    let supergraph_yaml_path = lsp_opts
+        .supergraph_yaml
+        .as_ref()
+        .and_then(|path| {
+            if path.is_relative() {
+                Some(
+                    Utf8PathBuf::try_from(std::env::current_dir().ok()?)
+                        .ok()?
+                        .join(path),
+                )
+            } else {
+                Some(path.clone())
+            }
+        })
+        .ok_or_else(|| anyhow!("Could not find supergraph.yaml file."))?;
+    let supergraph_content_root = supergraph_yaml_path.parent().unwrap().to_path_buf();
+
+    let studio_client = client_config.get_authenticated_client(&lsp_opts.plugin_opts.profile)?;
+
+    // Get the SupergraphConfig in a form we can use
+    let supergraph_config = SupergraphConfigResolver::default()
+        .load_remote_subgraphs(&studio_client, None)
+        .await?
+        .load_from_file_descriptor(
+            &mut stdin(),
+            Some(&FileDescriptorType::File(supergraph_yaml_path.clone())),
+        )?;
+    let lazily_resolved_supergraph_config = supergraph_config
+        .lazily_resolve_subgraphs(&supergraph_content_root)
+        .await?;
+
+    // Build the service to spin up the language server
+    let (service, socket, _receiver) = ApolloLanguageServer::build_service(
+        Config {
+            root_uri: supergraph_content_root.to_string(),
+            enable_auto_composition: false,
+            force_federation: false,
+            disable_telemetry: false,
+        },
+        HashMap::from_iter(
+            lazily_resolved_supergraph_config
+                .subgraphs()
+                .iter()
+                .map(|(a, b)| (a.to_string(), b.schema().clone())),
+        ),
+    );
+
+    let language_server = service.inner().clone();
+
+    // Spawn a separate thread to handle composition and passing that data to the language server
+    tokio::spawn(async move {
+        // Create a supergraph binary
+        // TODO: Check defaulting behaviour here and see if we need to centralise
+        let federation_version = lazily_resolved_supergraph_config
+            .federation_version()
+            .clone()
+            .unwrap_or(FederationVersion::LatestFedTwo);
+
+        // TODO: Let the supergraph binary exist inside its own task that can respond to being re-installed etc.
+        let supergraph_binary =
+            InstallSupergraph::new(federation_version.clone(), client_config.clone())
+                .install(
+                    None,
+                    lsp_opts.plugin_opts.elv2_license_accepter,
+                    lsp_opts.plugin_opts.skip_update,
+                )
+                .await?;
```

Comment:
One of the features (I think) of the old PR is that it allows people to update the version of Federation inside the `supergraph.yaml` while it's running. When doing the `dev` refactor we made an explicit decision that we didn't want to support that behaviour. Is that a dealbreaker for the LSP? Or could we ship this as is and see if there's sufficient demand after the fact to implement that?

Reply:
It's a pretty bad UX to not watch `supergraph.yaml` because the way to manually restart the language server underneath VS Code is really clunky—and it's not at all obvious when you should do that. If someone modifies their `supergraph.yaml` file and still sees out of date validations, it's going to be frustrating.